reporter_test.go

```diff
@@ -0,0 +1,29 @@
+package httpexpect
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+type mockT struct {
+	testing.T
+}
+
+var fatalfInvoked = false
+
+func (m *mockT) Fatalf(format string, args ...interface{}) {
+	fatalfInvoked = true
+}
+
+func TestFatalReporter(t *testing.T) {
+	mockBackend := &mockT{}
+	reporter := NewFatalReporter(mockBackend)
+
+	t.Run("Test Errorf", func(t *testing.T) {
+		reporter.Errorf("Test failed with backend: %v", mockBackend)
+		if !mockBackend.Failed() {
```

Comment:
Let's check fatalfInvoked instead. We want to know that Fatalf was invoked.

Reply:
We can remove `if !mockBackend.Failed() {` and just keep `assert.True(t, fatalfInvoked)`.